app/Harvest/Importers/AbstractImporter.php

```diff
@@ -109,7 +109,7 @@ protected function processHasMany(Model $model, $field, array $relatedRows, $all
         foreach ($relatedRows as $relatedRow) {
             $data = $this->mappers[$field]->map($relatedRow);
             $conditions = $this->getConditions($field, $data);
-            $instance = $model->$field()->firstOrNew($conditions);
+            $instance = $model->$field()->binaryFirstOrNew($conditions);
             $instance->forceFill($data);
```

Comment:
I'd like to point out two concerns with this approach (which also apply to `whereBinary` below)

1. This now says that _all_ the matching is done in the "binary" form (not sure what the right term is). Is that desirable? Wouldn't it generate false-positives?
2. If we sometimes do want the "strict" (i.e. not binary?) matching, we don't have a way to do it.

I'd suggest chaining the `where` conditions instead of build up the big `$conditions` array. I believe it would generate same results and allow us for some more flexibility. It could also be more specific to the problem at hand -- matching names.

Something like this (untested):

```suggestion
            $instance = $model->$field()->query()
            foreach ($conditions as $column => $value) {
                // Match name without diacritics
                if ($column === 'name') {
                    $instance->whereRaw("BINARY $column", $value);
                    continue;
                }
                $instance->where($column, $value);
            }
            $instance->binaryFirstOrNew()->forceFill($data);
```

what do you think?

Reply:
The accent/case-insenstivity works for use-cases like user search but here I think, on the contrary, it should be a strict comparison as you basically build a unique key that represents the importing record. Those data (ID's and such) usually don't change in time so if there is a 'change' it should cause creation of a new record.

Reply:
Oh, I misunderstood what `BINARY` does. Sorry & thanks.

Reply:
Still though, I think I'd still prefer the

```php
foreach ($conditions as $column => $value) {
    $instance->whereRaw("BINARY $column", $value);
}
```

instead of the `whereBinary` macro definition, as that seems mighty complicated. AFAIK the for loop should be just as performant and is easier to digest.

Reply:
@eronisko Sure, I'm okay with a simpler approach, this is a way better, thanks!

My original thought was to convert all tables to an accent-sensitive collation as we use Elasticsearch for full-text searches. We could still opt-in for different collation per use case. That's out of scope of this task though, but would be great to consider and discuss further.

Also we're currently using `utf8_slovak_ci` everywhere and since the database contains data of various languages it would certainly need at least a general collation.

Reply:
Thanks for the explanation, I wasn't aware of all this context. Looking forward to talking about it more